--- libpcap-1.3.0/configure.in.orig	2012-03-29 15:42:04.000000000 +0200
+++ libpcap-1.3.0/configure.in	2012-06-25 17:23:24.365976729 +0200
@@ -1275,10 +1275,7 @@
 	AC_DEFINE(PCAP_SUPPORT_USB, 1, [target host supports USB sniffing])
 	USB_SRC=pcap-usb-linux.c
 	AC_MSG_RESULT(yes)
-	ac_usb_dev_name=`udevinfo -q name -p /sys/class/usb_device/usbmon 2>/dev/null`
-	if test $? -ne 0 ; then
-	  ac_usb_dev_name="usbmon"
-	fi
+	ac_usb_dev_name="usbmon"
 	AC_DEFINE_UNQUOTED(LINUX_USB_MON_DEV, "/dev/$ac_usb_dev_name", [path for device for USB sniffing])
 	AC_MSG_NOTICE(Device for USB sniffing is /dev/$ac_usb_dev_name)
 	#
@@ -1406,7 +1403,7 @@
 		[
 		    AC_DEFINE(PCAP_SUPPORT_CANUSB, 1, [target host supports canusb])
 		    CANUSB_SRC=pcap-canusb-linux.c
-		    LIBS="-lusb-1.0 $LIBS"
+		    LIBS="-lusb-1.0 -lpthread $LIBS"
 		    AC_MSG_NOTICE(canusb sniffing is supported)
 		],
 		AC_MSG_NOTICE(canusb sniffing is not supported; install libusb1.0 lib devel to enable it)
