--- libpcap_0_5rel2/Makefile.in.wiget	Thu Jul 13 16:09:22 2000
+++ libpcap_0_5rel2/Makefile.in	Thu Jul 13 16:10:54 2000
@@ -126,19 +126,19 @@
 	$(CC) $(CFLAGS) -c $(srcdir)/bpf/net/bpf_filter.c -o bpf_filter.o
 
 install: force
-	$(INSTALL) -m 444 -o bin -g bin libpcap.a $(DESTDIR)$(LIBDEST)/libpcap.a
+	$(INSTALL) -m 444 libpcap.a $(DESTDIR)$(LIBDEST)/libpcap.a
 	$(RANLIB) $(DESTDIR)$(LIBDEST)/libpcap.a
 
 install-incl: force
-	$(INSTALL) -m 444 -o bin -g bin $(srcdir)/pcap.h \
+	$(INSTALL) -m 444 $(srcdir)/pcap.h \
 	    $(DESTDIR)$(INCLDEST)/pcap.h
-	$(INSTALL) -m 444 -o bin -g bin $(srcdir)/pcap-namedb.h \
+	$(INSTALL) -m 444 $(srcdir)/pcap-namedb.h \
 	    $(DESTDIR)$(INCLDEST)/pcap-namedb.h
-	$(INSTALL) -m 444 -o bin -g bin $(srcdir)/bpf/net/bpf.h \
+	$(INSTALL) -m 444 $(srcdir)/bpf/net/bpf.h \
 	    $(DESTDIR)$(INCLDEST)/net/bpf.h
 
 install-man: force
-	$(INSTALL) -m 444 -o bin -g bin $(srcdir)/pcap.3 \
+	$(INSTALL) -m 444 $(srcdir)/pcap.3 \
 	    $(DESTDIR)$(MANDEST)/man3/pcap.3
 
 clean:
