diff -urN libpcap-1.6.1.org/pcap-bt-monitor-linux.c libpcap-1.6.1/pcap-bt-monitor-linux.c
--- libpcap-1.6.1.org/pcap-bt-monitor-linux.c	2014-07-03 00:12:49.000000000 +0200
+++ libpcap-1.6.1/pcap-bt-monitor-linux.c	2014-07-20 16:53:44.000000000 +0200
@@ -39,7 +39,6 @@
 
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/hci.h>
-#include <bluetooth/mgmt.h>
 
 #include "pcap/bluetooth.h"
 #include "pcap-int.h"
@@ -47,6 +46,16 @@
 #define BT_CONTROL_SIZE 32
 #define INTERFACE_NAME "bluetooth-monitor"
 
+#ifndef MGMT_HDR_SIZE
+#define	MGMT_HDR_SIZE 6
+
+struct mgmt_hdr {
+        uint16_t opcode;
+        uint16_t index;
+        uint16_t len;
+} __packed;
+#endif
+
 int
 bt_monitor_findalldevs(pcap_if_t **alldevsp, char *err_str)
 {
