--- libpcap-0.8.1/configure.in.orig	2003-12-15 02:42:24.000000000 +0100
+++ libpcap-0.8.1/configure.in	2004-01-03 00:34:59.334440864 +0100
@@ -32,13 +32,11 @@
 AC_CHECK_FUNCS(ether_hostton strerror strlcpy)
 
 needsnprintf=no
-LIBOBJS=
 AC_CHECK_FUNCS(vsnprintf snprintf,,
 	[needsnprintf=yes])
 if test $needsnprintf = yes; then
-	LIBOBJS="$LIBOBJS snprintf.o"
+	AC_LIBOBJ([snprintf])
 fi
-AC_SUBST(LIBOBJS)
 
 dnl to pacify those who hate protochain insn
 AC_MSG_CHECKING(if --disable-protochain option is specified)
