--- libpcap-1.8.1/configure.ac.orig	2015-07-05 13:45:40.844741072 +0200
+++ libpcap-1.8.1/configure.ac	2015-07-05 14:14:49.988000998 +0200
@@ -458,9 +458,9 @@
 
                 incdir=-I/usr/include/libnl3
                 libnldir=
-                if test x$withval != x ; then
-                  libnldir=-L${withval}/lib/.libs
-                  incdir=-I${withval}/include
+                if test x$with_libnl != x -a x$with_libnl != xyes; then
+                  libnldir=-L${with_libnl}/lib/.libs
+                  incdir=-I${with_libnl}/include
                 fi
 
 		#
