--- libpcap-0.4a6.orig/inet.c
+++ libpcap-0.4a6/inet.c
@@ -69,7 +69,7 @@
 #endif
 
 /* Not all systems have IFF_LOOPBACK */
-#ifdef IFF_LOOPBACK
+#if defined(IFF_LOOPBACK) || defined(__linux__)
 #define ISLOOPBACK(p) ((p)->ifr_flags & IFF_LOOPBACK)
 #else
 #define ISLOOPBACK(p) ((p)->ifr_name[0] == 'l' && (p)->ifr_name[1] == 'o' && \
--- libpcap-0.4a6.orig/nametoaddr.c
+++ libpcap-0.4a6/nametoaddr.c
@@ -283,7 +283,7 @@
 	e = ep = (u_char *)malloc(6);
 
 	while (*s) {
-		if (*s == ':')
+		if (*s == ':' || *s == '.')
 			s += 1;
 		d = xdtoi(*s++);
 		if (isxdigit(*s)) {
